File: AVS3/circle.py
from dataclasses import dataclass
from math import pi

from .color import Color
from .point import Point

@dataclass
class Circle:
    '''
    Хранит информацию об окружности
    '''

    color: Color
    center: Point
    radius: int

    @staticmethod
    def read(buffer) -> 'Circle':
        color = Color.read(buffer)
        buffer.skip_whitespaces()
        center = Point.read(buffer)
        buffer.skip_whitespaces()
        radius = buffer.read_int()
        return Circle(color, center, radius)

    def perimiter(self):
        return 2 * pi * self.radius

    def __str__(self) -> str:
        return (f'Circle: color={self.color}, '
                f'center={self.center}, '
                f'radius={self.radius}')

File: lua/circle.lua
Color = require "color"
Point = require "point"

local Circle = {}
Circle.__index = Circle

function Circle:read(buffer)
	local result = setmetatable({}, self)
	result.color = Color:read(buffer)
	result.center = Point:read(buffer)
	result.radius = buffer:read_int()
	return result
end

function Circle:show()
	return string.format("Circle: color=%s, center=(%d, %d), radius=%d",
	                     self.color, self.center.x, self.center.y, self.radius)
end

function Circle:perimiter()
	return 2 * math.pi * self.radius
end

return Circle
